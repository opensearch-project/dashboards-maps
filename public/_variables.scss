/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

$mapHeaderOffset: 154px;
