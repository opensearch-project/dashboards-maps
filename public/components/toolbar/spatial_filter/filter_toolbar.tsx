/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
import {
  DRAW_FILTER_POLYGON,
  DRAW_FILTER_POLYGON_DEFAULT_LABEL,
  DRAW_FILTER_RECTANGLE,
  DRAW_FILTER_RECTANGLE_DEFAULT_LABEL,
  DrawFilterProperties,
  FILTER_DRAW_MODE,
} from '../../../../common';
import { FilterByShape } from './filter_by_shape';
import polygon from '../../../images/polygon.svg';

interface SpatialFilterToolBarProps {
  setFilterProperties: (properties: DrawFilterProperties) => void;
  mode: FILTER_DRAW_MODE;
}

export const SpatialFilterToolbar = ({ setFilterProperties, mode }: SpatialFilterToolBarProps) => {
  return (
    <EuiFlexGroup responsive={false} direction="column" alignItems="center" gutterSize="none">
      <EuiFlexItem>
        <FilterByShape
          setDrawFilterProperties={setFilterProperties}
          mode={mode}
          shapeMode={FILTER_DRAW_MODE.RECTANGLE}
          shapeLabel={DRAW_FILTER_RECTANGLE}
          defaultLabel={DRAW_FILTER_RECTANGLE_DEFAULT_LABEL}
          iconType={'vector'}
        />
      </EuiFlexItem>
      <EuiFlexItem>
        <FilterByShape
          setDrawFilterProperties={setFilterProperties}
          mode={mode}
          shapeMode={FILTER_DRAW_MODE.POLYGON}
          shapeLabel={DRAW_FILTER_POLYGON}
          defaultLabel={DRAW_FILTER_POLYGON_DEFAULT_LABEL}
          iconType={polygon}
        />
      </EuiFlexItem>
    </EuiFlexGroup>
  );
};
