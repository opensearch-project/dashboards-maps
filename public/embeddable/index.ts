/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

export * from './map_embeddable';
export * from './map_embeddable_factory';
