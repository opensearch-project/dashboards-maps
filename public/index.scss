/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

// Override EuiCollapsibleNavGroup background color
.layerConfigPanel {
  .euiCollapsibleNavGroup {
    background: transparent;

    // Remove any background color on hover if it exists
    &:hover {
      background: transparent;
    }

    // Ensure the content area is also transparent
    .euiCollapsibleNavGroup__content {
      background: transparent;
    }
  }
}
