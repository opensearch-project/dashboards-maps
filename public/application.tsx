/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import ReactDOM from 'react-dom';
import { AppMountParameters } from '../../../src/core/public';
import { MapServices } from './types';
import { MapsDashboardsApp } from './components/app';
import { OpenSearchDashboardsContextProvider } from '../../../src/plugins/opensearch_dashboards_react/public';

export const renderApp = (
  { element }: AppMountParameters,
  services: MapServices,
  dataSourceManagementEnabled: boolean
) => {
  ReactDOM.render(
    <OpenSearchDashboardsContextProvider services={services}>
      <MapsDashboardsApp />
    </OpenSearchDashboardsContextProvider>,
    element
  );

  return () => ReactDOM.unmountComponentAtNode(element);
};
