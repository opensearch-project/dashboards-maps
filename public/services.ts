/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { CoreStart } from '../../../src/core/public';
import { createGetterSetter } from '../../../src/plugins/opensearch_dashboards_utils/common';
import { TimefilterContract } from '../../../src/plugins/data/public';

export const postGeojson = async (
  requestBody: any,
  http: CoreStart['http'],
  dataSourceRefId: string
) => {
  try {
    const query = dataSourceRefId ? { dataSourceId: dataSourceRefId } : undefined;

    return await http.post('../api/custom_import_map/_upload', {
      body: requestBody,
      ...(query && { query }),
    });
  } catch (e) {
    return e;
  }
};

export const getIndex = async (
  indexName: string,
  http: CoreStart['http'],
  dataSourceRefId: string
) => {
  try {
    const query = dataSourceRefId ? { dataSourceId: dataSourceRefId } : undefined;
    return await http.post('../api/custom_import_map/_indices', {
      body: JSON.stringify({ index: indexName }),
      ...(query && { query }),
    });
  } catch (e) {
    return e;
  }
};

export const [getTimeFilter, setTimeFilter] = createGetterSetter<TimefilterContract>('TimeFilter');
