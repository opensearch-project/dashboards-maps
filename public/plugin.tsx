/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */
import { i18n } from '@osd/i18n';
import React from 'react';
import {
  AppMountParameters,
  CoreSetup,
  CoreStart,
  Plugin,
  PluginInitializerContext,
} from '../../../src/core/public';
import {
  AppPluginStartDependencies,
  MapServices,
  CustomImportMapPluginSetup,
  CustomImportMapPluginStart,
} from './types';
import {
  PLUGIN_NAME,
  MAPS_APP_ID,
  MAPS_APP_DISPLAY_NAME,
  PLUGIN_ID,
} from '../common/constants/shared';
import { ConfigSchema } from '../common/config';
import { AppPluginSetupDependencies } from './types';
import { RegionMapVisualizationDependencies } from '../../../src/plugins/region_map/public';
import { VectorUploadOptions } from './components/vector_upload_options';
import { OpenSearchDashboardsContextProvider } from '../../../src/plugins/opensearch_dashboards_react/public';
import {
  MAPS_APP_ICON,
  MAP_SAVED_OBJECT_TYPE,
  APP_PATH,
  MAPS_VISUALIZATION_DESCRIPTION,
} from '../common';
import { MapEmbeddableFactoryDefinition } from './embeddable';
import { setTimeFilter } from './services';

export class CustomImportMapPlugin
  implements Plugin<CustomImportMapPluginSetup, CustomImportMapPluginStart> {
  readonly _initializerContext: PluginInitializerContext<ConfigSchema>;
  constructor(initializerContext: PluginInitializerContext<ConfigSchema>) {
    this._initializerContext = initializerContext;
  }
  public setup(
    core: CoreSetup,
    { regionMap, embeddable, visualizations, dataSourceManagement }: AppPluginSetupDependencies
  ): CustomImportMapPluginSetup {
    const mapConfig: ConfigSchema = {
      ...this._initializerContext.config.get<ConfigSchema>(),
    };
    const dataSourceManagentEnabled: boolean = !!dataSourceManagement;
    // Register an application into the side navigation menu
    core.application.register({
      id: MAPS_APP_ID,
      title: MAPS_APP_DISPLAY_NAME,
      order: 5100,
      category: {
        id: 'opensearch',
        label: 'OpenSearch Plugins',
        order: 2000,
      },
      async mount(params: AppMountParameters) {
        // Load application bundle
        const { renderApp } = await import('./application');
        // Get start services as specified in opensearch_dashboards.json
        const [coreStart, depsStart] = await core.getStartServices();
        const {
          navigation,
          data,
          embeddable: useEmbeddable,
        } = depsStart as AppPluginStartDependencies;

        // make sure the index pattern list is up-to-date
        data.indexPatterns.clearCache();
        // make sure a default index pattern exists
        // if not, the page will be redirected to management and maps won't be rendered
        await data.indexPatterns.ensureDefaultIndexPattern();

        const services: MapServices = {
          ...coreStart,
          setHeaderActionMenu: params.setHeaderActionMenu,
          appBasePath: params.history,
          element: params.element,
          navigation,
          toastNotifications: coreStart.notifications.toasts,
          history: params.history,
          data,
          embeddable: useEmbeddable,
          scopedHistory: params.history,
          uiSettings: coreStart.uiSettings,
          mapConfig,
          dataSourceManagement,
          setActionMenu: params.setHeaderActionMenu,
        };
        params.element.classList.add('mapAppContainer');
        // Render the application
        return renderApp(params, services, dataSourceManagentEnabled);
      },
    });

    const mapEmbeddableFactory = new MapEmbeddableFactoryDefinition(async () => {
      const [coreStart, depsStart] = await core.getStartServices();
      const { navigation, data: useData } = depsStart as AppPluginStartDependencies;
      return {
        services: {
          ...coreStart,
          navigation,
          data: useData,
          toastNotifications: coreStart.notifications.toasts,
          mapConfig,
        },
      };
    });
    embeddable.registerEmbeddableFactory(MAP_SAVED_OBJECT_TYPE, mapEmbeddableFactory as any);

    visualizations.registerAlias({
      name: PLUGIN_ID,
      title: MAPS_APP_DISPLAY_NAME,
      description: MAPS_VISUALIZATION_DESCRIPTION,
      icon: MAPS_APP_ICON,
      aliasApp: MAPS_APP_ID,
      aliasPath: APP_PATH.CREATE_MAP,
      stage: 'production',
      appExtensions: {
        visualizations: {
          docTypes: [MAP_SAVED_OBJECT_TYPE],
          toListItem: ({ id, attributes, updated_at: updatedAt }) => ({
            description: attributes?.description,
            editApp: MAPS_APP_ID,
            editUrl: `${encodeURIComponent(id)}`,
            icon: MAPS_APP_ICON,
            id,
            savedObjectType: MAP_SAVED_OBJECT_TYPE,
            title: attributes?.title,
            typeTitle: MAPS_APP_DISPLAY_NAME,
            stage: 'production',
            updated_at: updatedAt,
          }),
        },
      },
    });

    const customSetup = async () => {
      const [coreStart] = await core.getStartServices();
      regionMap.addOptionTab({
        name: 'controls',
        title: i18n.translate('regionMap.mapVis.regionMapEditorConfig.controlTabs.controlsTitle', {
          defaultMessage: 'Import Vector Map',
        }),
        editor: (props: RegionMapVisualizationDependencies) => (
          <OpenSearchDashboardsContextProvider services={coreStart}>
            <VectorUploadOptions {...props} />
          </OpenSearchDashboardsContextProvider>
        ),
      });
    };
    customSetup();

    // Return methods that should be available to other plugins
    return {
      getGreeting() {
        return i18n.translate('mapsDashboards.greetingText', {
          defaultMessage: 'Hello from {name}!',
          values: {
            name: PLUGIN_NAME,
          },
        });
      },
    };
  }

  public start(core: CoreStart, { data }: AppPluginStartDependencies): CustomImportMapPluginStart {
    setTimeFilter(data.query.timefilter.timefilter);
    return {};
  }

  public stop() {}
}
