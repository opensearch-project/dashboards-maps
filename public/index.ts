/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import './index.scss';
import { PluginInitializerContext } from '../../../src/core/server';

import { CustomImportMapPlugin } from './plugin';
import { ConfigSchema } from './config';

// This exports static code and TypeScript types,
// as well as, OpenSearch Dashboards Platform `plugin()` initializer.
export function plugin(initializerContext: PluginInitializerContext<ConfigSchema>) {
  // @ts-ignore
  return new CustomImportMapPlugin(initializerContext);
}
export { CustomImportMapPluginSetup, CustomImportMapPluginStart } from './types';
