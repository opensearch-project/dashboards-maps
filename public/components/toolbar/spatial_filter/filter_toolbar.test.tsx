/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { shallow } from 'enzyme';
import React from 'react';
import { SpatialFilterToolbar } from './filter_toolbar';
import { FILTER_DRAW_MODE } from '../../../../common';
import { useOpenSearchDashboards } from '../../../../../../src/plugins/opensearch_dashboards_react/public';

// Mock the useOpenSearchDashboards hook
jest.mock('../../../../../../src/plugins/opensearch_dashboards_react/public');

describe('SpatialFilterToolbar tests', () => {
  beforeEach(() => {
    // Provide a default mock return value
    (useOpenSearchDashboards as jest.Mock).mockReturnValue({
      services: {
        uiSettings: {
          get: jest.fn().mockReturnValue(false), // default to light mode
        },
      },
    });
  });

  it('renders spatial filter before drawing', () => {
    const mockCallback = jest.fn();
    const toolbarComponent = shallow(
      <SpatialFilterToolbar setFilterProperties={mockCallback} mode={FILTER_DRAW_MODE.RECTANGLE} />
    );
    expect(toolbarComponent).toMatchSnapshot();
  });

  it('renders spatial filter while drawing', () => {
    const mockCallback = jest.fn();
    const toolbarComponent = shallow(
      <SpatialFilterToolbar setFilterProperties={mockCallback} mode={FILTER_DRAW_MODE.POLYGON} />
    );
    expect(toolbarComponent).toMatchSnapshot();
  });
});
