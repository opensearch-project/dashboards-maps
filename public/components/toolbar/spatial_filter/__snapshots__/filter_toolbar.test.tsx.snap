// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders spatial filter before drawing 1`] = `
<EuiFlexGroup
  alignItems="center"
  direction="column"
  gutterSize="none"
  responsive={false}
>
  <EuiFlexItem>
    <FilterByShape
      defaultLabel="rectangle"
      iconType="vector"
      setDrawFilterProperties={[MockFunction]}
      shapeLabel="Draw Rectangle"
      shapeMode="rectangle"
    />
  </EuiFlexItem>
  <EuiFlexItem>
    <FilterByShape
      defaultLabel="polygon"
      iconType={Object {}}
      setDrawFilterProperties={[MockFunction]}
      shapeLabel="Draw Polygon"
      shapeMode="polygon"
    />
  </EuiFlexItem>
</EuiFlexGroup>
`;

exports[`renders spatial filter while drawing 1`] = `
<EuiFlexGroup
  alignItems="center"
  direction="column"
  gutterSize="none"
  responsive={false}
>
  <EuiFlexItem>
    <FilterByShape
      defaultLabel="rectangle"
      iconType="vector"
      setDrawFilterProperties={[MockFunction]}
      shapeLabel="Draw Rectangle"
      shapeMode="rectangle"
    />
  </EuiFlexItem>
  <EuiFlexItem>
    <FilterByShape
      defaultLabel="polygon"
      iconType={Object {}}
      setDrawFilterProperties={[MockFunction]}
      shapeLabel="Draw Polygon"
      shapeMode="polygon"
    />
  </EuiFlexItem>
</EuiFlexGroup>
`;
